dbt/context/common.py

```diff
@@ -17,10 +17,16 @@
 def get_hooks(model, context, hook_key):
     hooks = model.get('config', {}).get(hook_key, [])
 
-    if isinstance(hooks, basestring):
+    if not isinstance(hooks, (list, tuple)):
         hooks = [hooks]
 
-    return hooks
+    wrapped_hooks = []
+    for hook in hooks:
+        if isinstance(hook, dict):
+            hook = json.dumps(hook)
```

Comment:
this is the only part of this branch that doesn't feel right to me. it seems like we should do the opposite: parse all the hooks to their dictionary version and return those. then you don't need to decode each hook individually in a loop before you can perform filter logic.

that would let you write, for example:

```
{% macro run_hooks(hooks, inside_transaction=True) %}
  {% for hook in hooks | selectattr('transaction', 'equalto', inside_transaction)  %}
    {% call statement(auto_begin=inside_transaction) %}
      {{ hook.get('sql') }}
    {% endcall %}
  {% endfor %}
{% endmacro %}
```

(and, in general, lets you do more powerful comprehensions on hooks)